features/auth/signup/Signup.tsx

```diff
@@ -178,21 +179,32 @@ export default function Signup() {
 
   useEffect(() => {
     (async () => {
-      if (urlToken) {
+      if (activationUID && activationToken) {
         setLoading(true);
         try {
-          authActions.updateSignupState(
-            await service.auth.signupFlowEmailToken(urlToken)
-          );
+          await service.auth.activateUser(activationUID, activationToken);
+          authActions.updateSignupState({
+            needBasic: false,
+            needAccount: true,
+            needAcceptCommunityGuidelines: true,
+            needFeedback: true,
+            needVerifyEmail: false,
+            flowToken: "token",
+          });
         } catch (err) {
           Sentry.captureException(err, {
             tags: {
               component: "auth/signup/Signup",
             },
           });
-          authActions.authError(
-            isGrpcError(err) ? err.message : t("global:error.fatal_message")
-          );
+          const errorMessages = isHttpError(err)
+            ? [
+                ...Object.values(err.errors || {}).flat(),
+                ...(err.error_messages || []),
+                t("global:error.fatal_message"),
+              ]
+            : [t("global:error.fatal_message")];
+          authActions.authError(errorMessages[0]);
```

Comment:
1. Currently there are two mutually exclusive errors responses returned by the API: `"Invalid user id or user doesn't exist."` and `"Invalid token for given user."`. So I think just showing the first (only) error message works here.
2. Yup, good call. I thought there was a case where we don't get `errors` but after double checking it is always returned along with `error_messages` for this endpoint.
3. `isHttpError` just checks if there is an `errors` or `error_messages` field even they're empty. So I think including `t("global:error.fatal_message")` covers us in that case.

Reply:
Couldn't we check if `errors` is empty, if it is then show fatal error, otherwise show the `errors` itself and reduce the need to duplicate fatal message and wrapping them both in a list?

Reply:
Nevermind, there actually is a case with no `errors`:
```
{"error_messages":["Stale token for given user."],"status_code":403}
```

Reply:
Right, we should update the typing for `errors` to be optional (with the `?` before `:`) then so it forces the code to check for it first, which I can see you did in the old code with the `||` fallback

Reply:
Pushed another change to (I think) address this